python-lib/openweathermap_utils/utils.py

```diff
@@ -0,0 +1,184 @@
+import logging
+from datetime import datetime, timedelta, timezone
+import pandas as pd
+from exceptions import OpenWeatherMapAPIError
+from requests import HTTPError
+import pwd
+import constants
+import os
+
+
+logger = logging.getLogger(__name__)
+
+
+def floor_time(dt=None, round_to='day'):
+    """
+    Floors the time to closest previous "round_to". If round_to = 'day', it return the day without time information
+    :param dt: Input datetime to floor
+    :param round_to: Granularity you want to floor on (day, hour, minute...)
+    :return: New time floored
+    """
+    datetime_els = ['year', 'month', 'day', 'hour', 'minute', 'second', 'microsecond']
+    if not round_to in datetime_els:
+        raise KeyError(f'Error in flooring dt: You have to choose a value between these: {datetime_els}')
+    rt_index = datetime_els.index(round_to)
+    return dt.replace(**{k: 0 if rt_index + i > 1 else 1 for i, k in enumerate(datetime_els[rt_index + 1:])})
+
+
+def datetime_to_timestamp(dt):
+    return int(dt.timestamp())
+
+
+def timestamp_to_datetime(ts):
+    return datetime.fromtimestamp(int(ts))
+
+
+def datetime_to_str(dt, formatter="%Y-%m-%d"):
+    return dt.strftime(formatter)
+
+
+def str_to_datetime(date_str, formatter="%Y-%m-%d"):
+    return datetime.strptime(date_str, formatter)
+
+
+def make_column_names_unique(df):
+    cols = pd.Series(df.columns)
+    for dup in cols[cols.duplicated()].unique():
+        cols[cols[cols == dup].index.values.tolist()] = [
+            dup + '_' + str(i) if i != 0 else dup for i in range(sum(cols == dup))
+        ]
+    df.columns = cols
+
+
+def flatten_dict_rec(el, names, new_dict):
+    if not el:
+        new_dict['.'.join(names)] = ''
+    if isinstance(el, dict):
+        for k, v in el.items():
+            flatten_dict_rec(v, names + [str(k)], new_dict)
+    elif isinstance(el, list):
+        for i, e in enumerate(el):
+            flatten_dict_rec(e, names + [str(i)], new_dict)
+    else:
+        new_dict['.'.join(names)] = el
+
+
+def flatten_dict(d):
```

Comment:
Need a comment to explain what this method does

Reply:
Fixed in 93397f3